pkg/services/queryhistory/database.go

```diff
@@ -41,6 +43,79 @@ func (s QueryHistoryService) createQuery(ctx context.Context, user *models.Signe
 	return dto, nil
 }
 
+func (s QueryHistoryService) searchQueries(ctx context.Context, user *models.SignedInUser, query SearchInQueryHistoryQuery) ([]QueryHistoryDTO, error) {
+	var queries []QueryHistoryDTO
+
+	if len(query.DatasourceUIDs) == 0 {
+		return nil, errors.New("no selected data source for query history search")
+	}
+
+	if query.Page <= 0 {
+		query.Page = 1
+	}
+
+	if query.Limit <= 0 {
+		query.Limit = 100
+	}
+
+	if query.Sort == "" {
+		query.Sort = "time-desc"
+	}
+
+	err := s.SQLStore.WithDbSession(ctx, func(session *sqlstore.DBSession) error {
+		sql := `SELECT
+			query_history.uid,
+			query_history.datasource_uid,
+			query_history.created_by,
+			query_history.created_at as "created_at",
+			query_history.comment,
+			query_history.queries,
+		`
+
+		if query.OnlyStarred {
+			sql = sql + `1 as "starred"
```

Comment:
I think you need to use [BooleanStr](https://github.com/grafana/grafana/blob/e814e7364bebe80c61c2439e9f54f908a7dcce87/pkg/services/sqlstore/migrator/dialect.go#L27) method here since there are differences between databases. For example https://github.com/grafana/grafana/blob/e814e7364bebe80c61c2439e9f54f908a7dcce87/pkg/services/libraryelements/database.go#L691

Reply:
Thanks for suggestion. Updated in https://github.com/grafana/grafana/pull/45932/commits/039372be4c92fe354e11540d684206e7afc14ff5